chapters/unitexpressions.tex

```diff
@@ -89,3 +89,171 @@ \section{The Syntax of Unit Expressions}\label{the-syntax-of-unit-expressions}
 \end{example}
+
+
+\section{Unit Checking}\label{unit-checking}
+
+How to verify that units are used in compatible ways is current not fully determined by the specification.
+This section gives rules for certain situations, but in general tools are expected to reason according to standard dimensional analysis.
+
+
+\subsection{Standard Dimensional Analysis}\label{standard-dimensional-analysis}
+
+This section gives an incomplete characterization of ``standard dimensional analysis'', also referred to as just \firstuse{dimensional analysis}.
+What is described below applies to unit checking in Modelica -- \emph{dimensional analysis} could have additional meanings in other contexts.
+While Modelica has a concept of \willintroduce{empty units} (described in later sections), standard dimensional analysis only deals with non-empty units such as \lstinline!"m"!, \lstinline!"m/s"!, or \lstinline!"1"!.
+It consists of two parts:
+\begin{itemize}
+\item
+  Unit compatibility requirements.
+\item
+  Rules for deriving the unit of an expression.
+\end{itemize}
+
+Examples of unit compatibility requirements that must be met in Modelica:
+\begin{itemize}
+\item
+  Dimensional homogeneity: The two sides of an equation or assignment statement must have the same unit.
+\item
+  The expression of a declaration equation must have the same unit as the component to which it belongs (special case of dimensional homogeneity).
+\item
+  The two operands of the additive operators \lstinline!+! and \lstinline!-! must have the same unit.
+\item
+  In a function call, the unit of an argument expression must match the unit (unless empty, see below) of the corresponding function input component.
+\end{itemize}
+
+% TODO: Replace these examples by giving unit semantics where operators and built-in functions are defined in the specification,
+% and just include some likes to places where such semantics are given.
+Examples of unit derivation in Modelica:
+\begin{itemize}
+\item
+  The result of additive operators has the same unit as the operands.
+\item
+  The result of multiplication has a unit obtained by multiplying the operands' units.
+\item
+  Built-in operators such as \lstinline!pre! and \lstinline!previous! preserve units, while \lstinline!der! changes the unit by dividing it by \lstinline!"s"!.
+\end{itemize}
+
+
+\subsection{Empty and Undefined Units}\label{empty-and-undefined-units}
+
+In situations where the specification does not prescribe how to determine the unit of an expression, the unit of that expression is said to be \firstuse[undefined unit]{undefined}.
+It is then not possible for a tool to reject the equation (or other construct) with support in the specification, and options for the tool include silently not performing unit checking, or applying checks based dimensional analysis.
```

Comment:
Above you "expect tools" to apply dimensional analysis, and here it is one of several options. This might be improved maybe (is it a shall, should, or may)?

Reply:
The _in general tools are expected_ wasn't meant to be a "shall", but to leave some room for interpretation while waiting for the specification to give a more complete set of rules.  How about resolving it like this?
> This section gives rules for certain situations, but in general tools should reason according to standard dimensional analysis.

Reply:
Sounds good.

Reply:
Done.  Resolving.